API/NoDocs/Microsoft.Extensions.Http.Resilience/HttpResiliencePipelineBuilderExtensions.cs

```diff
@@ -0,0 +1,13 @@
+// Assembly 'Microsoft.Extensions.Http.Resilience'
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.Compliance.Classification;
+
+namespace Microsoft.Extensions.Http.Resilience;
+
+public static class HttpResiliencePipelineBuilderExtensions
+{
+    public static IHttpResiliencePipelineBuilder SelectPipelineByAuthority(this IHttpResiliencePipelineBuilder builder, DataClassification classification);
```

Comment:
```suggestion
    public static IHttpResiliencePipelineBuilder SelectPipelineByAuthority(this IHttpResiliencePipelineBuilder builder);
```

Do we want to tie HttpResilience with redaction? Can the default implementation just return raw URLs. If anyone wants to do something like redacting the URLs they can use the other `SelectPipelineBy`.

This would allow us to have one less dependency in this package.

Reply:
We have `SelectPipelineByAuthority` that would just return raw authority without any redaction:

Then for more advanced scenarios where redaction is actually necessary the consumer can do:

``` csharp
services
    .AddHttpClient(ClientId)
    .AddStandardResilienceHandler()
    .SelectPipelineBy(serviceProvider =>
    {
        var redactor = serviceProvider.GetRequiredService<IRedactorProvider>().GetRedactor(DataClassification.Unknown);

        return request =>
        {
            var authority = request.RequestUri.GetLeftPart(UriPartial.Authority);
            return redactor.Redact(authority);
        };
    });
```

This thing is I never practically saw scenario where the authority need to be redacted. Usually it's something that within the path or query params.

Reply:
Done.